boat_simulator/nodes/physics_engine/kinematics_computation.py

```diff
@@ -0,0 +1,125 @@
+"""This module contains the kinematics computations for the boat."""
+
+from typing import Tuple
+from numpy.typing import ArrayLike
+from boat_simulator.common.types import Scalar
+from boat_simulator.nodes.physics_engine.kinematics_data import KinematicsData
+import numpy as np
+import boat_simulator.common.constants as constants
+
+
+class BoatKinematics:
+    # Private class member default
+    __relative_data = KinematicsData()
+    __global_data = KinematicsData()
+
+    def __init__(self, timestep: Scalar, mass: Scalar, inertia: ArrayLike) -> None:
+        self.__timestep = timestep
+        self.__boat_mass = mass
+        self.__inertia = np.array(inertia, dtype=np.float32)
+        self.__inertia_inverse = np.linalg.inv(self.__inertia)
+
+    # Updates the kinematics data and returns the kinematics data objects
+    def step(self, net_force: ArrayLike, net_torque: ArrayLike
+             ) -> Tuple[KinematicsData, KinematicsData]:
+
+        next_relative_acceleration, next_relative_velocity = \
+            self.__update_linear_relative_data(net_force)
+
+        yaw_radians = self.__update_ang_data(net_torque)
```

Comment:
I feel that swapping the order of these function calls makes the code slightly more readable since the global data update follows. What do you think?